internal/agent/device/lifecycle/manager.go

```diff
@@ -92,15 +99,39 @@ func (m *LifecycleManager) Initialize(ctx context.Context, status *v1alpha1.Devi
 }
 
 func (m *LifecycleManager) Sync(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
-	// TODO: implement
+	if current.Decommission == nil && desired.Decommission != nil {
+		m.decommission = &DecommissionRequest{
+			request: desired.Decommission,
+			state:   v1alpha1.DecommissionStateStarted,
+		}
+	}
 	return nil
 }
 
 func (m *LifecycleManager) AfterUpdate(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
 	// TODO: implement
+	// we can use this to execute pre-key/cert wipe actions based on the decommission target type,
+	// once we support more types
+	if m.decommission != nil {
+		m.decommission.state = v1alpha1.DecommissionStateComplete
+	}
 	return nil
 }
 
+// point of no return - wipes management cert and keys
+func (m *LifecycleManager) WipeAndReboot() {
+	err := os.Remove(m.managementCertPath)
```

Comment:
For anything touching the filesystem, we want to use the `fileio` package. So, in this case, you will want to add `fileio.Writer` as a dependency and call `RemoveFile`. This is to allow us to perform mock test against the file system inside a tmpDir.